addons/web/controllers/main.py

```diff
@@ -706,7 +706,7 @@ def restore(self, master_pwd, backup_file, name, copy=False):
             dispatch_rpc('db', 'restore', [master_pwd, name, data, str2bool(copy)])
             return http.local_redirect('/web/database/manager')
         except Exception, e:
-            error = "Database restore error: %s" % e
+            error = "Database restore error: %s" % repr(e)
```

Comment:
Replacing %s by %r in the string would have the same effect and be a smaller change.